storage/index_insert_queue.go

```diff
@@ -216,11 +219,13 @@ func (q *nsIndexInsertQueue) Stop() error {
 	default:
 		// Loop busy, already ready to consume notification
 	}
+	// FOLLOWUP(prateek): we need to wait until this queue is drained before
+	// we can correctly return success/failure.
```

Comment:
Is it tough to add this with the current PR?  I worry we'll forget about this.

Reply:
Nah it's a minor change but I'd prefer to do in another PR. Opened https://github.com/m3db/m3db/issues/593 to track. Will put up a change in a few.